pkg/didcomm/protocol/outofband/service.go

```diff
@@ -0,0 +1,341 @@
+	logger.Debugf("receive inbound message : %s", msg)
+
+	if !s.Accept(msg.Type()) {
+		return "", fmt.Errorf("unsupported message type %s", msg.Type())
+	}
+
+	// TODO should request messages with no attachments be rejected?
+	//  https://github.com/hyperledger/aries-rfcs/issues/451
+
+	go func() {
+		s.ActionEvent() <- service.DIDCommAction{
+			ProtocolName: Name,
+			Message:      msg,
+			Continue:     continueFunc(s.callbackChannel, msg, myDID, theirDID),
+			Stop: func(e error) {
+				// TODO noop - nothing to do here (not even cleanup)
+			},
+			Properties: nil,
+		}
+	}()
+
+	return "", nil
+}
+
+// HandleOutbound handles outbound messages
+func (s *Service) HandleOutbound(_ service.DIDCommMsg, _, _ string) error {
+	// TODO implement
+	return errors.New("not implemented")
+}
+
+func continueFunc(c chan *callback, msg service.DIDCommMsg, myDID, theirDID string) func(interface{}) {
+	return func(_ interface{}) {
+		c <- &callback{
+			msg:      msg,
+			myDID:    myDID,
+			theirDID: theirDID,
+		}
+	}
+}
+
+func listener(
+	callbacks chan *callback,
+	didEvents chan service.StateMsg,
+	handleReqFunc func(*callback) error,
+	handleDidEventFunc func(msg service.StateMsg) error) func() {
+	return func() {
+		for {
+			select {
+			case c := <-callbacks:
+				// TODO add support for handling the 'invitation' message
+				switch c.msg.Type() {
+				case RequestMsgType:
+					err := handleReqFunc(c)
+					if err != nil {
+						logutil.LogError(logger, Name, "handleRequestCallback", err.Error(),
+							logutil.CreateKeyValueString("msgType", c.msg.Type()),
+							logutil.CreateKeyValueString("msgID", c.msg.ID()))
+					}
+				default:
+					logutil.LogError(logger, Name, "callbackChannel", "unsupported msg type",
+						logutil.CreateKeyValueString("msgType", c.msg.Type()),
+						logutil.CreateKeyValueString("msgID", c.msg.ID()))
+				}
+			case e := <-didEvents:
+				err := handleDidEventFunc(e)
+				if err != nil {
+					logutil.LogError(logger, Name, "handleDidEvent", err.Error())
+				}
+			}
+		}
+	}
+}
+
+func (s *Service) handleRequestCallback(c *callback) error {
+	// TODO either transform to a didexchange.invitation object or refactor didexchange.Service to accept this object
+	connID, err := s.didSvc.HandleInbound(c.msg, c.myDID, c.theirDID)
+	if err != nil {
+		return fmt.Errorf("didexchange service failed to handle inbound request : %w", err)
+	}
+
+	req := &Request{}
+
+	err = c.msg.Decode(req)
+	if err != nil {
+		return fmt.Errorf("failed to decode request message : %w", err)
+	}
+
+	// TODO if we want to implement retries then we should be saving state before invoking
+	//  the didexchange service
+	err = s.save(&myState{
+		ID:           c.msg.ID(),
+		ConnectionID: connID,
+		Request:      req,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to save my state : %w", err)
+	}
+
+	return nil
+}
+
+func (s *Service) handleDidEvent(e service.StateMsg) error {
```

Comment:
handleDidEvent should be handleDIDEvent.

Reply:
done

(this should be a linter check!)

Reply:
https://github.com/hyperledger/aries-framework-go/issues/87